pkg/controller/controller.go

```diff
@@ -110,26 +119,45 @@ type controller struct {
 	brokerLister           listers.BrokerLister
 	serviceClassLister     listers.ServiceClassLister
 	instanceLister         listers.InstanceLister
+	bindingLister          listers.BindingLister
+	brokerQueue            workqueue.RateLimitingInterface
+	serviceClassQueue      workqueue.RateLimitingInterface
+	instanceQueue          workqueue.RateLimitingInterface
+	bindingQueue           workqueue.RateLimitingInterface
 }
 
 // Run runs the controller until the given stop channel can be read from.
-func (c *controller) Run(stopCh <-chan struct{}) {
+func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	defer runtimeutil.HandleCrash()
+	defer func() {
+		c.brokerQueue.ShutDown()
+		c.serviceClassQueue.ShutDown()
+		c.instanceQueue.ShutDown()
+		c.bindingQueue.ShutDown()
+	}()
+
 	glog.Info("Starting service-catalog controller")
 
+	for i := 0; i < workers; i++ {
+		go wait.Until(worker(c.brokerQueue, "Broker", c.reconcileBrokerKey), time.Second, stopCh)
+		go wait.Until(worker(c.serviceClassQueue, "ServiceClass", c.reconcileServiceClassKey), time.Second, stopCh)
+		go wait.Until(worker(c.instanceQueue, "Instance", c.reconcileInstanceKey), time.Second, stopCh)
+		go wait.Until(worker(c.bindingQueue, "Binding", c.reconcileBindingKey), time.Second, stopCh)
+	}
+
 	<-stopCh
 	glog.Info("Shutting down service-catalog controller")
 }
 
 // Broker handlers and control-loop
 
 func (c *controller) brokerAdd(obj interface{}) {
-	broker, ok := obj.(*v1alpha1.Broker)
-	if broker == nil || !ok {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
```

Comment:
Nit: I think it would be extremely helpful to newcomers to indicate why we use this particular key func in the first place we use it (which is here) within this file.